pkg/chartutil/chartfile.go

```diff
@@ -42,7 +42,16 @@ func LoadChartfile(filename string) (*chart.Metadata, error) {
 //
 // 'filename' should be the complete path and filename ('foo/Chart.yaml')
 func SaveChartfile(filename string, cf *chart.Metadata) error {
+	// Pull out the dependencies of a v1 Chart, since there's
```

Comment:
You can exclude fields in the serializer but it is at definition time and would therefore exclude the field from all serializing (e.g. `Dependencies []*Dependency json:"dependencies,omitempty"` --> `Dependencies []*Dependency`). This seems best option as it is just for the v1 case and not in all cases.

Reply:
Maybe update the comment

Reply:
I'm not sure what you want changed in the comment. Would it be clearer to say "there's no way to tell the serialiser to conditionally skip a field"?

Reply:
> > requirements.yaml was being unconditionally merged into Chart.yaml, which produced packages which failed helm lint.
> 
> I am unable to reproduce this. Can you give more details about it?

With Helm v2.16.2:
```powershell
helm create testchart
```

Add a requirements.yaml:
```yaml
dependencies:
  - name: magic-ip-address
    version: ">0"
    repository: "@stable"
```

With Helm v2.16.2:
```powershell
helm dependency update testchart
```

With Helm v3.0.0 or v3.0.1
```powershell
helm package testchart
helm lint testchart-0.1.0.tgz
```

This will reproduce the failure:
```
==> Linting .\testchart-0.1.0.tgz
[INFO] Chart.yaml: icon is recommended
[ERROR] Chart.yaml: dependencies are not valid in the Chart file with apiVersion 'v1'. They are valid in apiVersion 'v2'   
Error: 1 chart(s) linted, 1 chart(s) failed
```

Reply:
Maybe something like this: "no way to tell the serialiser to skip a field" --> "no way to tell the serialiser to skip a field for just this use case"